### PR TITLE
Globalfixes next138 v2

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
This patchset is composed of a number of changes that affect the whole project in one step. This includes:

- Namespace changes for TPM types
- Adding of const qualifiers to SAPI functions
- Changing the return code of almost all functions

 **New in version 2**
- Fixed false removal of rc-check
- Changed notation on byte buffer construction
see review on #615

### CODE REVIEW

File: test/unit/TPML-marshal.c [hunk 1]
```diff
@@ -43,20 +42,18 @@ tpml_marshal_success(void **state)
...
-    assert_int_equal (ptr2->pcrSelections[0].pcrSelect[1], 0xbb);
-    assert_int_equal (ptr2->pcrSelections[0].pcrSelect[2], 0xcc);
-    assert_int_equal (ptr2->pcrSelections[1].hash, HOST_TO_BE_16(TPM_ALG_SHA256));
-    assert_int_equal (ptr2->pcrSelections[1].sizeofSelect, 2);
-    assert_int_equal (ptr2->pcrSelections[1].pcrSelect[0], 0xdd);
-    assert_int_equal (ptr2->pcrSelections[1].pcrSelect[1], 0xee);
+
+    uint8_t expect [] = {
+        0, 0, 0, 2,
+        (uint8_t)(TPM_ALG_SHA1 / 256), (uint8_t)TPM_ALG_SHA1,
```
Comment (tstruk) [inline test/unit/TPML-marshal.c]: Have you pushed the correct version? I don't see any difference between v1 and v2 here.

**Comment by tstruk** [comment]
Just as a general note. We are trying to unify the coding standards across TSS code. Currently the code is not consistent. I have added a file describing the formatting we want to adopt. I have just added a new coding standards requirements to our CONTRIBUTING.md file. Please take it into account.

File: test/unit/TPML-marshal.c (hunk 1 above)
Comment (AndreasFuchsSIT) [reply test/unit/TPML-marshal.c]: Yep you are right...

Some screw-up on rebase merge-conflicts... Resending...

**Comment by AndreasFuchsSIT** [comment]
Sent updated pull request on #623

**Pull request closed without merging**